File: getNumberOfPlants/getNumberOfPlants.js
function getNumberOfPlants(fieldSize, unit, crop) {

  let pickedUnit;

  if (unit === "acre" || unit === "acres") {
    pickedUnit = 4046.86;
  } else if (unit === "hectare" || unit === "hectares") {
    pickedUnit = 10000;
  }

  const allCrop = {
    corn: 1,
    wheat: 0.1,
    soybeans: 0.5,
    tomatoes: 0.25,
    lettuce: 0.2
  };

  let pickedCrop = allCrop[crop];
  let totalFieldSize = fieldSize * pickedUnit;

  return Math.floor(totalFieldSize / pickedCrop);
}

File: getNumberOfPlants/get_number_of_plants.py
import math

def get_number_of_plants(field_size, unit, crop):
    if unit in ('acre', 'acres'):
        pickedUnit = 4046.86
    elif unit in ('hectare', 'hectares'):
        pickedUnit = 10000
    else:
        return None

    allCrop = {
        "corn": 1,
        "wheat": 0.1,
        "soybeans": 0.5,
        "tomatoes": 0.25,
        "lettuce": 0.2
    }

    pickedCrop = allCrop.get(crop)
    if pickedCrop is None:
        return None

    totalFieldSize = field_size * pickedUnit
    return math.floor(totalFieldSize / pickedCrop)
